src/map/collection.cpp

```diff
@@ -0,0 +1,243 @@
+			return 0;
+		}
+
+		collection->GroupID = static_cast<uint16>(constant);
+	}
+
+	if (!exists) {
+		this->put(item->nameid, collection);
+	}
+	return 1;
+}
+
+bool CollectionDatabase::reload(){
+	if( !TypesafeYamlDatabase::reload() ){
+		return false;
+	}
+
+	return true;
+}
+
+CollectionDatabase collection_db;
+
+/**
+ * Display the success/failure roulette wheel when trying to catch monster.
+ * @param sd : player requesting
+ * @param target_class : monster ID of monster to catch
+ * @return 0
+ */
+int collection_catch_process1(struct map_session_data *sd, t_itemid item_id)
+{
+	nullpo_ret(sd);
+
+	sd->catch_collection_class = item_id;
+	clif_catch_collection_process(sd);
+
+	return 0;
+}
+
+/**
+ * Begin the actual catching process of a monster.
+ * @param sd : player requesting
+ * @param target_id : monster ID of monster to catch
+ * @return 0:success, 1:failure
+ */
+int collection_catch_process2(struct map_session_data* sd, uint32 target_id)
+{
+	nullpo_retr(1, sd);
+
+	std::shared_ptr<s_collection_db> collection = collection_db.find(sd->catch_collection_class);
+
+	struct mob_data* md = (struct mob_data*)map_id2bl(target_id);
+
+	if (!md || md->bl.type != BL_MOB || md->bl.prev == NULL) {	// Invalid inputs/state, abort capture
+		clif_collection_roulette(sd, 0);
+		sd->catch_collection_class = COLLECTION_CATCH_FAIL;
+		sd->itemid = 0;
+		sd->itemindex = -1;
+		return 1;
+	}
+
+	// Monster ID not in array
+	if (!collection) {
+		clif_emotion(&md->bl, ET_ANGER);
+		clif_collection_roulette(sd, 0);
+		sd->catch_collection_class = COLLECTION_CATCH_FAIL;
+		return 1;
+	}
+
+	// If the target is a valid monster, we have a few exceptions
+	if (collection) {
+		if (sd->catch_collection_class == COLLECTION_CATCH_UNIVERSAL && !status_has_mode(&md->status, MD_STATUSIMMUNE)) {
+			sd->catch_collection_class = md->mob_id;
+		}
+		else if (sd->catch_collection_class == COLLECTION_CATCH_UNIVERSAL_ITEM && sd->itemid == collection->ConsumeID) {
+			sd->catch_collection_class = md->mob_id;
+		}
+	}
+
+	// Monster ID not in array
+	if (std::find(collection->MobID.begin(), collection->MobID.end(), md->mob_id) == collection->MobID.end()) {
+		clif_emotion(&md->bl, ET_ANGER);
+		clif_collection_roulette(sd, 0);
+		sd->catch_collection_class = COLLECTION_CATCH_FAIL;
+		return 1;
+	}
+
+	// Is it should use the same formula as monster catch rate?
+	int collection_catch_rate = (collection->CaptureRate+ (sd->status.base_level - md->level)*30 + sd->battle_status.luk*20)*(200 - get_percentage(md->status.hp, md->status.max_hp))/100;
+
+	if(collection_catch_rate < 1)
+		collection_catch_rate = 1;
+
+	if(rnd()%10000 < collection_catch_rate) {
+		unit_remove_map(&md->bl,CLR_OUTSIGHT);
+		status_kill(&md->bl);
+		clif_collection_roulette(sd,1);
+		itemdb_group.pc_get_itemgroup(collection->GroupID, 1, sd);
+	} else {
+		clif_collection_roulette(sd,0);
+		sd->catch_collection_class = COLLECTION_CATCH_FAIL;
```

Comment:
move this to before last return maybe. so just whatever the result it doesn't save last usage item_id

Reply:
Thanks @cydh, it was moved.